From 4a4cd9c7ed2157c4369e2c81e8eb973e0323f2ee Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 14 Jan 2016 09:59:52 +0800
Subject: [PATCH 0843/5242] MLK-12251 video: mipi_dsi: fix an unsigned
 compared against zero issue

commit  f7c62936124ad2469f02ecdd705439d80e5f1153 from
https://source.codeaurora.org/external/imx/linux-imx.git

Compared 'mux' against zero is meaningless. So correct this.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.c b/drivers/video/fbdev/mxc/mipi_dsi.c
index 3ebf25b..3e78a53 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi.c
@@ -790,7 +790,7 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 	struct resource *res;
 	u32 dev_id, disp_id;
 	const char *lcd_panel;
-	unsigned int mux;
+	int mux;
 	int ret = 0;
 
 	mipi_dsi = devm_kzalloc(&pdev->dev, sizeof(*mipi_dsi), GFP_KERNEL);
@@ -889,7 +889,7 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 	mux = mipi_dsi->bus_mux->get_mux(dev_id, disp_id);
 	if (mux >= 0)
 		regmap_update_bits(mipi_dsi->regmap, mipi_dsi->bus_mux->reg,
-				   mipi_dsi->bus_mux->mask, mux);
+				   mipi_dsi->bus_mux->mask, (unsigned int)mux);
 	else
 		dev_warn(&pdev->dev, "invalid dev_id or disp_id muxing\n");
 
-- 
1.7.9.5

